app/src/main/java/com/therap/amin/currencyconverter/Fragments/CurrencyConversionFragment.java

```diff
@@ -19,70 +17,69 @@
 
-    @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-        View view = inflater.inflate(R.layout.activity_main, container, false);
+    @InjectView(R.id.spnToCurrency)
+    Spinner toCurrencySpinner;
 
-        fileProcessor = new FileProcessor(getActivity());
-        sharedPreferences = getActivity().getSharedPreferences(Constants.PREFERENCE_KEY, Context.MODE_PRIVATE);
+    @Inject
+    FileProcessor fileProcessor;
 
-        sourceRadioGroup = (RadioGroup) view.findViewById(R.id.rgSource);
-        webRadioButton = (RadioButton) view.findViewById(R.id.rbWeb);
-        ownvalueRadioButton = (RadioButton) view.findViewById(R.id.rbOwn);
+    @InjectView(R.id.rgSource)
+    RadioGroup sourceRadioGroup;
 
-        sourceRadioGroup.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
-            @Override
-            public void onCheckedChanged(RadioGroup group, int checkedId) {
-                int id = sourceRadioGroup.getCheckedRadioButtonId();
+    @InjectView(R.id.rbWeb)
+    RadioButton webRadioButton;
 
-                if (id == webRadioButton.getId()) {
-                    selectedSource = "web";
-                } else if (id == ownvalueRadioButton.getId()) {
-                    selectedSource = "own";
-                }
-                updateUI();
-            }
-        });
+    @InjectView(R.id.rbOwn)
+    RadioButton ownvalueRadioButton;
 
+    String selectedSource = "web";
+    ArrayAdapter<String> inputCurrencyAdapter;
+    ArrayAdapter<String> outputCurrencyAdapter;
+    SharedPreferences sharedPreferences;
 
-        availableCurrencies = getResources().getStringArray(R.array.currencies);
-        tvOutputCurrencyValue = (TextView) view.findViewById(R.id.tvOutputCurrencyValue);
-        etInputCurrencyValue = (EditText) view.findViewById(R.id.etCurrencyAmount);
-        tvOutputCurrencyValue.setText("0");
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_currency_conversion, container, false);
+    }
 
-        fromCurrencySpinner = (Spinner) view.findViewById(R.id.spnFromCurrency);
-        toCurrencySpinner = (Spinner) view.findViewById(R.id.spnToCurrency);
+    public void onViewCreated(final View view, final Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
 
         inputCurrencyAdapter = new ArrayAdapter<String>(getActivity(), android.R.layout.simple_spinner_item, availableCurrencies);
         fromCurrencySpinner.setAdapter(inputCurrencyAdapter);
         outputCurrencyAdapter = new ArrayAdapter<String>(getActivity(), android.R.layout.simple_spinner_item, availableCurrencies);
         toCurrencySpinner.setAdapter(outputCurrencyAdapter);
 
+        sharedPreferences = getActivity().getSharedPreferences(Constants.PREFERENCE_KEY, Context.MODE_PRIVATE);
```

Comment:
You can also inject sharedPreference with Robojuice. Check https://github.com/roboguice/roboguice/wiki/Provided-Injections